File: string-paradigm-expression-original/src/main/java/org/jesperancinha/string/paradigm/original/expression/RegionImpl.java
package org.jesperancinha.string.paradigm.original.expression;

import com.jesperancinha.string.paradigm.api.Region;

public class RegionImpl extends Region {
	public RegionImpl(String type, String size) {
		this.regionName = type;
		this.assignedNumber = size;
	}

	public String getRegionName() {
		return regionName;
	}

	public String getAssignedNumber() {
		return assignedNumber;
	}

	public String getDescription() {
		return regionName + ", " + assignedNumber;
	}
}

File: string-paradigm-expression-original/src/main/kotlin/org/jesperancinha/string/paradigm/original/expression/RegionImpl.kt
package org.jesperancinha.string.paradigm.original.expression

import org.jesperancinha.string.paradigm.api.Region


class RegionImpl(regionName: String, assignedNumber: String) : Region(regionName, assignedNumber) {
    override var description: String
        get() = "$regionName, $assignedNumber"
        set(_) {}
}